pkg/logging/logger.go

```diff
@@ -21,6 +21,24 @@ func GetLogger(prefix string) *logrus.Entry {
 	return log.WithField("prefix", prefix)
 }
 
+// AddField add a field to an existing logrus.Entry
+func AddField(e *logrus.Entry, name string, value interface{}) *logrus.Entry {
+	l := log.WithField(name, value)
+	for k, v := range e.Data {
+		l = l.WithField(k, v)
+	}
+	return l
+}
+
+// AddFields adds multiple fields to an existing logrus.Entry
+func AddFields(e *logrus.Entry, fields map[string]interface{}) *logrus.Entry {
+	l := e
+	for k, v := range fields {
+		l = AddField(l, k, v)
+	}
+	return l
+}
```

Comment:
As it was already discussed before, I believe builtin `logrus` functions is a better solution.

Reply:
Yeah, sorry for déjà vu. It's totally fair to review one PR at at time and review the other one once the rebase has happened.